log/lvl.go

```diff
@@ -0,0 +1,377 @@
+func LLvlf2(f string, args ...interface{}) { lvlf(-2, f, args...) }
+
+// LLvlf3 *always* prints
+func LLvlf3(f string, args ...interface{}) { lvlf(-3, f, args...) }
+
+// LLvlf4 *always* prints
+func LLvlf4(f string, args ...interface{}) { lvlf(-4, f, args...) }
+
+// LLvlf5 *always* prints
+func LLvlf5(f string, args ...interface{}) { lvlf(-5, f, args...) }
+
+// TestOutput sets the DebugVisible to 0 if 'show'
+// is false, else it will set DebugVisible to 'level'
+//
+// Usage: TestOutput( test.Verbose(), 2 )
+func TestOutput(show bool, level int) {
+	debugMut.Lock()
+	defer debugMut.Unlock()
+
+	if show {
+		debugVisible = level
+	} else {
+		debugVisible = 0
+	}
+}
+
+// SetDebugVisible set the global debug output level in a go-rountine-safe way
+func SetDebugVisible(lvl int) {
+	debugMut.Lock()
+	defer debugMut.Unlock()
+	debugVisible = lvl
+}
+
+// DebugVisible returns the actual visible debug-level
+func DebugVisible() int {
+	debugMut.RLock()
+	defer debugMut.RUnlock()
+	return debugVisible
+}
+
+// SetShowTime allows for turning on the flag that adds the current
+// time to the debug-output
+func SetShowTime(show bool) {
+	debugMut.Lock()
+	defer debugMut.Unlock()
+	showTime = show
+}
+
+// ShowTime returns the current setting for showing the time in the debug
+// output
+func ShowTime() bool {
+	debugMut.Lock()
+	defer debugMut.Unlock()
+	return showTime
+}
+
+// SetUseColors can turn off or turn on the use of colors in the debug-output
+func SetUseColors(show bool) {
+	debugMut.Lock()
+	defer debugMut.Unlock()
+	useColors = show
+}
+
+// UseColors returns the actual setting of the color-usage in log
+func UseColors() bool {
+	debugMut.Lock()
+	defer debugMut.Unlock()
+	return useColors
+}
+
+// MainTest can be called from TestMain to set up some default
+// variables
+func MainTest(m *testing.M) {
+	flag.Parse()
+	TestOutput(testing.Verbose(), 3)
+	code := m.Run()
+	AfterTest(nil)
+	os.Exit(code)
+}
+
+// ParseEnv looks at the following environment-variables:
+// - DEBUG_LVL - for the actual debug-lvl - default is 1
+// - DEBUG_TIME - whether to show the timestamp - default is false
+// - DEBUG_COLOR - whether to color the output - default is true
+func ParseEnv() {
+	var err error
+	dv := os.Getenv("DEBUG_LVL")
+	if dv != "" {
+		debugVisible, err = strconv.Atoi(dv)
+		Lvl3("Setting level to", dv, debugVisible, err)
+		if err != nil {
+			Error("Couldn't convert", dv, "to debug-level")
+		}
+	}
+	dt := os.Getenv("DEBUG_TIME")
+	if dt != "" {
+		showTime, err = strconv.ParseBool(dt)
+		Lvl3("Setting showTime to", dt, showTime, err)
+		if err != nil {
+			Error("Couldn't convert", dt, "to boolean")
+		}
+	}
+	dc := os.Getenv("DEBUG_COLOR")
+	if dc != "" {
+		useColors, err = strconv.ParseBool(dc)
+		Lvl3("Setting useColor to", dc, showTime, err)
+		if err != nil {
+			Error("Couldn't convert", dc, "to boolean")
+		}
+	}
+}
+
+// AddFlags adds the flags and the variables for the debug-control
+func AddFlags() {
```

Comment:
Maybe `RegisterFlags` ? 
Also, it'd be nice to specify that it is using the official  `flags` package